libs/cohere/langchain_cohere/chat_models.py

```diff
@@ -410,6 +410,8 @@ def _get_generation_info(self, response: NonStreamedChatResponse) -> Dict[str, A
             )
         if hasattr(response, "token_count"):
             generation_info["token_count"] = response.token_count
+        elif hasattr(response.meta, "tokens"):
+            generation_info["token_count"] = response.meta.tokens
```

Comment:
I think this suggestion may be helpful:
* response.meta.tokens may be None
* returning a dict instead of the pydantic model seems helpful
```suggestion
        elif hasattr(response.meta, "tokens") and response.meta.tokens is not None:
            generation_info["token_count"] = response.meta.tokens.dict()
```